Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -828,39 +838,49 @@ void LoadEventNexus::loadEvents(API::Progress *const prog,
   std::string classType = monitors ? "NXmonitor" : "NXevent_data";
   ::NeXus::Info info;
   bool oldNeXusFileNames(false);
-  bool hasTotalCounts(true);
   bool haveWeights = false;
   auto firstPulseT = DateAndTime::maximum();
-  while (true) { // should be broken when entry name is set
-    const auto entry = m_file->getNextEntry();
-    const std::string entry_name(entry.first);
-    const std::string entry_class(entry.second);
-    if (entry_name == NULL_STR && entry_class == NULL_STR)
-      break;
-
-    if (entry_class == classType) {
-      // open the group
-      m_file->openGroup(entry_name, classType);
-
-      if (takeTimesFromEvents) {
-        /* If we are here, we are loading logs, but have failed to establish
-         * the run_start from the proton_charge log. We are going to get this
-         * from our event_time_zero instead
-         */
-        auto localFirstLast = firstLastPulseTimes(*m_file, this->g_log);
-        firstPulseT = std::min(firstPulseT, localFirstLast.first);
-      }
-      // get the number of events
-      std::size_t num = numEvents(*m_file, hasTotalCounts, oldNeXusFileNames);
-      bankNames.emplace_back(entry_name);
-      bankNumEvents.emplace_back(num);
 
-      // Look for weights in simulated file
-      haveWeights = exists(*m_file, "event_weight");
+  const std::map<std::string, std::set<std::string>> &allEntries =
+      descriptor->getAllEntries();
+
+  auto itClassEntries = allEntries.find(classType);
+
+  if (itClassEntries != allEntries.end()) {
 
-      m_file->closeGroup();
+    const std::set<std::string> &classEntries = itClassEntries->second;
+    const std::regex classRegex("(/" + m_top_entry_name + "/)([^/]*)");
+    std::smatch groups;
+
+    for (const std::string &classEntry : classEntries) {
+
+      if (std::regex_match(classEntry, groups, classRegex)) {
+        const std::string entry_name(groups[2].str());
+        m_file->openGroup(entry_name, classType);
+
+        if (takeTimesFromEvents) {
+          /* If we are here, we are loading logs, but have failed to establish
+           * the run_start from the proton_charge log. We are going to get this
+           * from our event_time_zero instead
+           */
+          auto localFirstLast = firstLastPulseTimes(*m_file, this->g_log);
+          firstPulseT = std::min(firstPulseT, localFirstLast.first);
+        }
+        // get the number of events
+        const std::string prefix = "/" + m_top_entry_name + "/" + entry_name;
```

Comment:
consider std::string::append when concatenating 3 or more strings
https://clang.llvm.org/extra/clang-tidy/checks/performance-inefficient-string-concatenation.html

Reply:
Thanks for sharing (nice link BTW), I have trouble connecting these concepts. The link info applies to left-values in which the string (thus underlying buffer) already exist. This line is applying RAII since it's concatenating into a right-value that is assigned to a newly created string. I can switch to `const std::string prefix("/" + m_top_entry_name + "/" + entry_name);` to make it more explicit. As for performance (at least with GNU `basic_string`) each `std::string` carries a stack allocated char[16] (15 + null char) for SSO. Using RAII and knowing the size upfront might be using this buffer if length <= 15.  Thoughts?

Reply:
I **don't** see any benefit switching to `const std::string prefix("/" + m_top_entry_name + "/" + entry_name);`

I'm curious about the benefits of doing something like 

```suggestion
        const std::string prefix = std::string("/").append(m_top_entry_name).append("/").append(entry_name);
```
but don't expect a measurable performance difference unless the strings are much longer than 15 characters.